aspnetcore/migration/70-80.md

```diff
@@ -73,64 +73,298 @@ In the project file, update each [Microsoft.AspNetCore.*](https://www.nuget.org/
+
+1. Rename `App.razor` in the `Client` project to `Routes.razor`.
+
+1. In `Routes.razor`, update the value of the `AppAssembly` attribute to `typeof(Program).Assembly`.
+
+1. Add an entry to the `_Imports.razor` file to make shorthand render modes available to the app:
+
+   ```razor
+   @using static Microsoft.AspNetCore.Components.Web.RenderMode
+   ```
+
+1. Make a copy of the `_Imports.razor` file and add it to the `Server` project.
+
+1. Make the following changes to the `App.razor` file:
+
+   Replace the `<title>` tag with the `HeadOutlet` component. Start by removing the `<title>` tag: 
+
+   ```diff
+   - <title>...</title>
+   ```
+
+   Add the `HeadOutlet` component at the end of the `<head>` content with the interactive WebAssembly render mode and prerendering disabled:
+
+   ```razor
+   <HeadOutlet @rendermode="new InteractiveWebAssemblyRenderMode(prerender: false)" />
+   ```
+
+   Replace the following `div` tag with the `Routes` component using the interactive WebAssembly render mode and prerendering disabled:
+   
+   ```diff
+   - <div id="app">
+   -     ...
+   - </div>
+   ```
+
+   ```razor
+   <Routes @rendermode="new InteractiveWebAssemblyRenderMode(prerender: false)"
+   ```
+
+   Update the `blazor.webassembly.js` script to `blazor.web.js`:
+
+   ```diff
+   - <script src="_framework/blazor.webassembly.js"></script>
+   + <script src="_framework/blazor.web.js"></script>
+   ```
+
+1. Remove the following lines from `Client/Program.cs`:
+
+   ```diff
+   - builder.RootComponents.Add<App>("#app");
+   - builder.RootComponents.Add<HeadOutlet>("head::after");
+   ```
+   
+1. Update `Server/Program.cs`:
+
+   Add Razor component and interactive WebAssembly component services to the project. Call `AddRazorComponents` with a chained call to `AddInteractiveWebAssemblyComponents`:
+
+   ```csharp
+   builder.Services.AddRazorComponents()
+       .AddInteractiveWebAssemblyComponents();
+   ```
+
+   Add [Antiforgery Middleware](xref:blazor/security/index#antiforgery-support) to the request processing pipeline immediately after the call to `app.UseRouting`. If there are calls to `app.UseRouting` and `app.UseEndpoints`, the call to `app.UseAntiforgery` must go between them. Calls to `app.UseAntiforgery` must be placed after calls to `app.UseAuthentication` and `app.UseAuthorization`.
+
+   ```csharp
+   app.UseAntiforgery();
+   ```
+
+   Remove the following line:
+   
+   ```diff
+   - app.UseBlazorFrameworkFiles();
+   ```
+
+   Remove the following line and replace it with a call to `MapRazorComponents`, supplying the `App` component as the root component type, and add a chained call to `AddInteractiveWebAssemblyRenderMode`:
+   
+   ```diff
+   - app.MapFallbackToFile("index.html");
+   ```
 
-For more information, see the following content in the *Data binding* article:
+   ```csharp
+   app.MapRazorComponents<App>()
+       .AddInteractiveWebAssemblyRenderMode();
+   ```
```

Comment:
```suggestion
   ```csharp
   app.MapRazorComponents<App>()
       .AddInteractiveWebAssemblyRenderMode()
       .AddAdditionalAssemblies(typeof(Counter).Assembly);
   ```
```

Reply:
Didn't they go with something like ...

```
.AddAdditionalAssemblies(typeof(BlazorServer80.Client._Imports).Assembly);
```

... i.e., not the `Counter` component.

Reply:
https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Program.cs#L118

Reply:
I see the check prior to that if sample components are added that they do go with the `Counter` component. However ... if the dev then deletes this because this is a project update, they'll hose this line if they go with the `Counter` component, so it seems safer to go with the `_Imports` file.

Reply:
Yeah, it just needs to be some type that lives in the client project. It could be the client Program class, and component, etc.